tests/native_api_tests/test_api_model.cpp

```diff
@@ -829,6 +842,130 @@ TEST_CASE("API Model") {
                                  PowerGridRegularError);
         }
     }
+
+    SUBCASE("Test get indexer") {
+        std::vector<ID> const node_id{1, 2, 3};
+        std::vector<double> const node_u_rated{10.0e3, 10.0e3, 10.0e3};
+
+        DatasetConst input_dataset{"input", false, 1};
+        input_dataset.add_buffer("node", std::ssize(node_id), std::ssize(node_id), nullptr, nullptr);
```

Comment:
narrowing cast issue here. the alternatives are:
* a single `auto const n_nodes = static_cast<Idx>(node_id.size());` and similar for every other type
* hardcode the values
* use `static_cast<Idx>(node_id.size())` and equivalent everywhere
* add an overload specifically for signed types combined with `narrow_cast<Idx>` in the implementation

thought this might be the nicest way but LMK if you have a different opinion

Reply:
This is nice, no need to change it. But I would have though that `std::size` should be enough and handle it with implicit conversion to `Idx`. This then leads me to the question: Should we handle this in the function signature itself, similar to the above implementation for the contructor of the Dataset?

Reply:
nope, cause `Idx` is signed, so `std::size` leads to narrowing conversion `size_t` to `Idx`. this is an issue that the C++ board has recognized and is very hard to change (breaking many projects), so they decided that introducing `std::ssize` is a good workaround.

If at some point we get sick of using `std::ssize`, we can then extend the C++ API, but the difference between C++ API and C API only introduces more confusing, rather than less, when it doesn't really matter either way

Reply:
basically `std::ssize(xx_container)` is the same in our context as `(Idx)xx_container.size()`.